src/app.rs

```diff
@@ -66,46 +74,61 @@ impl BdiffApp {
             sett
         };
 
+        let started_with_arguments = !paths.is_empty();
+
         let mut ret = Self {
             next_hv_id: 0,
             hex_views,
             settings,
+            started_with_arguments,
             ..Default::default()
         };
 
-        if !paths.is_empty() {
-            for path in paths {
-                let _ = ret.open_file(path);
+        log::info!("Loading project config from file");
+        let config_path = Path::new("bdiff.json");
+
+        let config = if started_with_arguments {
+            let file_configs = paths
+                .into_iter()
+                .map(|a| a.into())
+                .collect::<Vec<FileConfig>>();
+
+            Config {
+                files: file_configs,
+                changed: true,
             }
+        } else if config_path.exists() {
+            read_json_config(config_path).unwrap()
         } else {
-            log::info!("Loading project config from file");
-            let config_path = Path::new("bdiff.json");
-
-            if config_path.exists() {
-                let config = read_json_config(config_path).unwrap();
+            let conf = Config::default();
+            write_json_config(config_path, &conf).expect("Failed to write empty project config");
```

Comment:
I think we don't necessarily want to write a config on app launch. It might get annoying having new files on disk when users aren't expecting it. I think it's okay for the setting stored globally but could be a bit much for the directory-specific config file

Reply:
You do make a great point!

Especially if we're going to be switching to multiple per-project configurations, this has to go.